### ISSUE TITLE
[Feature Request] Closing backticks after opening a quotation via backticks

### ISSUE THREAD

**Issue by Megamiun** [telegramdesktop/tdesktop#7877]
### Is your feature request related to a problem?
At the moment, when we wish to open a code block/quotation, we have to write three backticks, write the text and then close it.

It is not a issue per se, but when looking at the user experience using Slack, that we just write ``` and it already creates a quotation/code section, it always bothers me a little, even more so that to do that, I would have to click `shift + '` 12 times, as my keyboard only writes one backtick when pressing this combination 2 times.

It makes the process very tedious and clumsy, and any help would make it better.

### Describe the solution you'd like
After adding three backticks, the cursor would skip to the next line and add new backticks closing the section.

### Describe alternatives you've considered
There is too a solution, using the mouse and doing a right lick on the selected text, but other than that I see none.

**Comment by AndydeCleyre** [comment]
FWIW I was looking at the [GFM](https://github.github.com/gfm/#fenced-code-blocks) and [CommonMark](https://spec.commonmark.org/0.29/#fenced-code-blocks) specs, and saw:

> If the end of the containing block (or document) is reached and no closing code fence has been found, the code block contains all of the lines after the opening code fence until the end of the containing block (or document).

This is identical in both those specs.

[Testing at Babelmark 2](https://johnmacfarlane.net/babelmark2/?text=Is+the+rest+of+this+a+code+block%3F%0A%0A%60%60%60%0Aone%0Atwo%0Athree) indicates that behavior is implemented by at least the following libraries:

- commonmark.js 0.29.0
- cheapskate 0.1.0.5
- MD4C 0.2.7
- MD4C (strict) 0.2.7
- league/commonmark 1.3.0
- cebe/markdown GFM 1.2.0
- cebe/markdown MarkdownExtra 1.2.0
- markdig 0.18.1.0
- Maruku 0.7.3.beta1
- Maruku (Math-Enabled) 0.7.3.beta1
- s9e\TextFormatter (Fatdown/PHP) 2.7.0
- MultiMarkdown 6 6.3.0
- Parsedown 1.6.0
- markdown-it 11.0.0

Another workaround would be to use an external snippet expander like [espanso](https://espanso.org/), with a config like:

````yml
backend: Clipboard
fast_inject: false
matches:
  - trigger: :mono
    replace: |
      ```
      $|$
      ```
````

**Comment by AndydeCleyre** [comment]
Parsing content from an opening fence to the end of the message as a code block (if no closing fence is found) would still be a valuable improvement for user convenience and markdown compliance.

@Megamiun would you consider that to solve this issue, even though it is not the auto-insertion of the closing fence that you suggested?

**Comment by Megamiun** [comment]
I agree, it's a valuable improvement. I will see if I can get some time to do it ASAP.

**Comment by AndydeCleyre** [comment]
This is still relevant.

Whether or not auto completion of the fence is added, this would improve the situation and increase markdown spec compliance:

> Parsing content from an opening fence to the end of the message as a code block (if no closing fence is found)

I'll also note an alternative not mentioned here yet (I think): use the keyboard to select text, then use shortcut to toggle mono (ctrl+shift+m on Linux).
